src/cluster.py

```diff
@@ -45,7 +45,14 @@ def add_node_to_cluster(self, event: ops.framework.EventBase) -> None:
         """Add node to microceph cluster."""
         if not event.unit:
             return
-        cmd = ["microceph", "cluster", "add", event.unit.name]
+        # get hostname using unit name.
+        hostnames = self.charm.peers.get_all_unit_values(
+            key=event.unit.name, include_local_unit=True
+        )
+        if not hostnames:
+            event.defer()
+
+        cmd = ["microceph", "cluster", "add", hostnames[0]]
```

Comment:
Can we rely on the new node always being the first in the list?

Reply:
we have to hardcode hostnames[0] here as `peers.get_all_unit_values` always returns a list. Since we use a unit name (coming from the event) we get a list of one element (or an empty list) . Since we defer the event if `hostnames` is empty, we can always use hostnames[0].